services/brig/src/Brig/API/User.hs

```diff
@@ -606,19 +606,21 @@ mkActivationKey (ActivatePhone p) = do
 -- Password Management
 
 changePassword :: UserId -> PasswordChange -> ExceptT ChangePasswordError AppIO ()
-changePassword u cp = do
-    activated <- lift $ Data.isActivated u
+changePassword uid cp = do
+    activated <- lift $ Data.isActivated uid
     unless activated $
         throwE ChangePasswordNoIdentity
-    currpw <- lift $ Data.lookupPassword u
+    currpw <- lift $ Data.lookupPassword uid
     let newpw = cpNewPassword cp
     case (currpw, cpOldPassword cp) of
-        (Nothing,        _) -> lift $ Data.updatePassword u newpw
+        (Nothing,        _) -> lift $ Data.updatePassword uid newpw
         (Just  _,  Nothing) -> throwE InvalidCurrentPassword
         (Just pw, Just pw') -> do
             unless (verifyPassword pw' pw) $
                 throwE InvalidCurrentPassword
-            lift $ Data.updatePassword u newpw >> revokeAllCookies u
+            when (verifyPassword newpw pw) $
```

Comment:
Would it make sense to make `checkNewIsDifferent` top-level and call it here? Maybe pass in the type of error to call? Or is it small enough to not matter?

Reply:
I'd say it's borderline.  I will leave it as is this time and try to lean more towards smaller functions in the future.